tests/dbfs/test_api.py

```diff
@@ -89,15 +102,26 @@ def dbfs_api():
 
 
 class TestDbfsApi(object):
+    def test_list_files_recursive(self, dbfs_api):
+        json = {
+            'files': [TEST_FILE_JSON0, TEST_DIR_JSON, TEST_FILE_JSON1]
+        }
+        dbfs_api.client.list.return_value = json
+        files = dbfs_api.list_files("dbfs:/")
+
+        assert len(files) == 2
+        assert TEST_FILE_INFO0 == files[0]
+        assert TEST_FILE_INFO1 == files[1]
+
```

Comment:
There doesn't seem to be a test for a real recursive call.

The variable `TEST_FILE_JSON2` is unused. When you make a test that uses it and does a real recursive call, I expect it to fail for the reason I commented about above; the mismatch between `FileInfo` and `DbfsPath`.